packages/hardhat-core/src/internal/util/global-dir.ts

```diff
@@ -75,3 +87,29 @@ export async function getCompilersDir() {
   await fs.ensureDir(compilersCache);
   return compilersCache;
 }
+
+/**
+ * Checks if the user has given (or refused) consent for telemetry.
+ *
+ * Returns undefined if it can't be determined.
+ */
+export function hasConsentedTelemetry(): boolean | undefined {
```

Comment:
Wouldn't the whole pr be simpler if this were async?

Reply:
Yes, but that would make the error reporter async, and since it's used in the constructor of `provider/node.ts`, it would mean having to find a way around that.

Reply:
Oh, it's because of the error reporter! I thought it was because of the project creation module, which is already async, so that confused me.